skimage/color/colorlabel.py

```diff
@@ -182,6 +182,10 @@ def _label2rgb_overlay(label, image=None, colors=None, alpha=0.3,
         # Opacity doesn't make sense if no image exists.
         alpha = 1
     else:
+        if label.ndim > 2:
+            raise ValueError("label2rgb only supports 2D inputs :"
+                             " label and image shape (M,N) and (M,N,3)")
+
```

Comment:
Actually, it looks like a tiny change below could just make this function work for nD instead!

Can you please remove this change and instead replace the 2-line if statement below with:
```Python
        if not image.shape[:label.ndim] == label.shape:
            raise ValueError("`image` and `label` must be the same shape")

        if not image.shape[-1] == 3:
            raise ValueError(
                "`image` must be RGB (image.shape[-1] must be 3)."
            )
```

Reply:
I think just changing `image.shape[:2]` to `image.shape[:label.ndim]` in the above is enough to get nD working here, but we will need at least one new test case to verify this.

Reply:
Sir, I have updated code to support nD images and added test for it. Doc has been updated too. Thank you, sir, for pointing out the code. Please review changes, if anything needed do mention.